Mu2eKinKal/inc/DOCAStrawHitUpdater.hh

```diff
@@ -0,0 +1,40 @@
+//
+// Simple updater of StrawHits based on (unbiased) Distance Of Closest Approach (DOCA)
+//
+#ifndef Mu2eKinKal_DOCAStrawHitUpdater_hh
+#define Mu2eKinKal_DOCAStrawHitUpdater_hh
+#include "KinKal/Detector/WireHitStructs.hh"
+
+namespace mu2e {
+// Update based just on (unbiased) DOCA to the wire, not including this hit
+  class DOCAStrawHitUpdater {
+    public:
+      DOCAStrawHitUpdater() : maxdoca_(-1.0), mindoca_(0), maxddoca_(-1) {}
+      DOCAStrawHitUpdater(double maxdoca, double minddoca, double maxddoca) :
+        maxdoca_(maxdoca), mindoca_(minddoca), maxddoca_(maxddoca) {}
+     // set the state based on the current DOCA value
+      KinKal::WireHitState wireHitState(double doca) const;
+      auto maxDOCA() const { return maxdoca_; }
+      auto minDOCA() const { return mindoca_; }
+      auto maxDriftDOCA() const { return maxddoca_; }
+    private:
+      double maxdoca_; // maximum DOCA to still use a hit
+      double mindoca_; // minimum DOCA to use drift information
+      double maxddoca_; // maximum DOCA to use drift information
+  };
+
+  KinKal::WireHitState DOCAStrawHitUpdater::wireHitState(double doca) const {
+    using KinKal::WireHitState;
+    WireHitState whstate(WireHitState::inactive);
+    double absdoca = fabs(doca);
+    if( absdoca < maxdoca_){ // hit isn't too far from the wire
+      if(absdoca > mindoca_ && absdoca < maxddoca_){  // in the sweet spot: use the DOCA to sign the ambiguity
```

Comment:
is the state supposed to be null or inactive when doca > maxdoca?

Reply:
Inactive.

Reply:
Maybe you're asking about maxdoca < doca < maxddoca?  Those are set to null, as inaccurate but correctly associated hits.  In practice I haven't used that category, but it could be explored.

Reply:
Ah I missed the distinction of maxdoca_ vs maxddoca_, thanks